### PR TITLE
DR-2965 Allow moving self hosted files for predictable id datasets

### PR DESCRIPTION
**Pull request by nmalfroy**
It turns out, we already support this so I added an explicit integration test that ingests then re-ingests with a different source path

Note: to use this, you do need to ingest with the same load tag as the file that was originally ingested and the file needs to have the same size and MD5 hash (or else the fileids would be different)

I also made a couple of small drive by changes:
- ~~Make sure that we properly store the parent directory for root level directories in Firestore (Fixes an issue where you can't enumerate the top level directories)~~ (I reverted this since it caused all sorts of other problems and created a ticket - [DR-3005](https://broadworkbench.atlassian.net/browse/DR-3005) - to address this
- Add some more accurate logging in the bulk loader

[DR-3005]: https://broadworkbench.atlassian.net/browse/DR-3005?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by samanehsan — APPROVED** [review]
This looks good to me! Thanks for including the drive-by changes too! 🎉

File: src/main/java/bio/terra/service/filedata/FileMetadataUtils.java [hunk 1]
```diff
@@ -22,6 +22,8 @@ public class FileMetadataUtils {
 
   public FileMetadataUtils() {}
 
+  // TODO: this currently returns the directory as "" if you pass in a one-level deep item
+  // https://broadworkbench.atlassian.net/browse/DR-3005 is to fix (changing breaks tests badly)
```
Comment (okotsopoulos) [inline src/main/java/bio/terra/service/filedata/FileMetadataUtils.java:26]: Bummer that this drive-by change didn't work, but I appreciate that you left a trail of breadcrumbs with this comment!

File: src/test/java/bio/terra/service/filedata/FileMetadataUtilsTest.java [hunk 2]
```diff
@@ -205,6 +205,29 @@ public void extractDirectoryPathsTest() {
     assertThrows(IllegalArgumentException.class, () -> FileMetadataUtils.extractDirectoryPaths(""));
   }
 
+  @Test
+  public void pathParsingTest() {
+    assertThat("empty string returns empty", FileMetadataUtils.getDirectoryPath(""), equalTo(""));
+    assertThat("root directory dir looks ok", FileMetadataUtils.getDirectoryPath("/"), equalTo(""));
+    assertThat("root directory file looks ok", FileMetadataUtils.getName("/"), equalTo(""));
+
+    // It's admitedly strange that this is what we expect but changing the behavior causes untold
```
Comment (okotsopoulos) [inline src/test/java/bio/terra/service/filedata/FileMetadataUtilsTest.java:214]: :elmofire:

**Review by okotsopoulos — APPROVED** [review]
Looks good!
